### PR TITLE
JunOS Security Policies Global

### PR DESCRIPTION
**Pull request by pranav-bhatt**
This PR adds the security_policies_global module, which is responsible for managing the following security policy parameters:
- default_policy
- policy_rematch
- policy_stats
- pre_id_default_policy
- traceoptions

This module complements the JunOS Security Policies module, which allows defining of firewall and global security policies.

### CODE REVIEW

**Comment by pranav-bhatt** [comment]
Recheck

**Comment by komaldesai13** [comment]
recheck

**Comment by pranav-bhatt** [comment]
recheck

**Comment by pranav-bhatt** [comment]
recheck

**Comment by komaldesai13** [comment]
depends on  https://github.com/ansible-network/windmill-config/pull/938#issue-1175047662

**Comment by komaldesai13** [comment]
recheck

**Comment by goneri** [comment]
recheck

File: plugins/modules/junos_security_policies_global.py [hunk 1]
```diff
@@ -0,0 +1,979 @@
...
+        by executing the command B(show security policies).
+      - The state I(parsed) reads the configuration from C(running_config) option and
+        transforms it into Ansible structured data as per the resource module's argspec
+        and the value is then returned in the I(parsed) key within the result.
+    type: str
+  state:
+    choices:
+    - merged
+    - replaced
+    - overridden
```
Comment (rohitthakur2590) [inline plugins/modules/junos_security_policies_global.py:193]: We can mention that replaced and overridden will work in similar way

File: plugins/modules/junos_security_policies_global.py [hunk 2]
```diff
@@ -0,0 +1,979 @@
...
+#         "no_remote_trace": true
+#     }
+# }
+#
+#
+"""
+RETURN = """
+before:
+  description: The configuration prior to the module execution.
+  returned: when state is I(merged), I(replaced), I(overridden), I(deleted) or I(purged)
```
Comment (rohitthakur2590) [inline plugins/modules/junos_security_policies_global.py:893]: we don't have purged for this

File: plugins/modules/junos_security_policies_global.py [hunk 3]
```diff
@@ -0,0 +1,979 @@
...
+after:
+  description: The resulting configuration after module execution.
+  returned: when changed
+  type: dict
+  sample: >
+    This output will always be in the same format as the
+    module argspec.
+commands:
+  description: The set of commands pushed to the remote device.
+  returned: when state is I(merged), I(replaced), I(overridden), I(deleted) or I(purged)
```
Comment (rohitthakur2590) [inline plugins/modules/junos_security_policies_global.py:907]: puged is not supported

File: plugins/modules/junos_security_policies_global.py [hunk 4]
```diff
@@ -0,0 +1,979 @@
...
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.junipernetworks.junos.plugins.module_utils.network.junos.argspec.security_policies_global.security_policies_global import (
+    Security_policies_globalArgs,
+)
+from ansible_collections.junipernetworks.junos.plugins.module_utils.network.junos.config.security_policies_global.security_policies_global import (
+    Security_policies_global,
+)
+
+
+def main():
```
Comment (rohitthakur2590) [inline plugins/modules/junos_security_policies_global.py:963]: Please add functionality to check for empty config for opearation it is required.,
also test needs to be added for that

**Review by rohitthakur2590 — CHANGES_REQUESTED** [review]

**Comment by goneri** [comment]
recheck

**Comment by pranav-bhatt** [comment]
recheck

**Comment by pranav-bhatt** [comment]
recheck

**Comment by rohitthakur2590** [comment]
recheck

**Review by rohitthakur2590 — APPROVED** [review]

**Pull request merged**